stdlib/public/core/Integers.swift.gyb

```diff
@@ -2361,8 +2395,7 @@ ${operatorComment('&' + x.operator, True)}
 //===----------------------------------------------------------------------===//
 
 /// An integer type that can represent only nonnegative values.
-public protocol UnsignedInteger : BinaryInteger
-  where Magnitude : BinaryInteger { }
+public protocol UnsignedInteger : BinaryInteger { }
```

Comment:
Yep! Constraints on associated types can be recursive as long as the associated type is declared in a higher protocol.